src/main/java/org/folio/edge/sip2/repositories/FolioResourceProvider.java

```diff
@@ -79,44 +94,51 @@ public FolioResourceProvider(String okapiUrl, String tenant, Vertx vertx) {
   }
 
   @Override
-  public Future<JsonObject> editResource(IRequestData fromData) {
+  public Future<IResource> editResource(IRequestData fromData) {
     return null;
   }
 
   @Override
-  public Future<JsonObject> deleteResource(IRequestData resource) {
+  public Future<IResource> deleteResource(IRequestData resource) {
     return null;
   }
 
   private void setHeaders(
       Map<String, String> headers,
-      HttpRequest<Buffer> request) {
+      HttpRequest<Buffer> request,
+      SessionData sessionData) {
     for (Map.Entry<String,String> entry : Optional.ofNullable(headers)
         .orElse(Collections.emptyMap()).entrySet()) {
       request.putHeader(entry.getKey(), entry.getValue());
     }
 
-    if (okapiToken != null) {
-      request.putHeader(HEADER_X_OKAPI_TOKEN, okapiToken);
+    if (sessionData != null) {
```

Comment:
This code is due to the default implementation of `IRequestData.getSessionData()` returning `null`. Some tests don't set this, I think, which is why I put it in. It is admittedly bad. I wanted to make it easier to implement `IRequestData` via a lambda/functional interface in the tests, since we don't really need `SessionData` in most cases.

Now that you call me out on this practice, I think I am being too lazy here. I'll refactor this so that `sessionData` cannot be `null`.